python/pyspark/sql/pandas/conversion.py

```diff
@@ -452,24 +457,27 @@ def _create_from_pandas_with_arrow(self, pdf, schema, timezone):
                 struct.add(name, from_arrow_type(field.type), nullable=field.nullable)
             schema = struct
 
-        # Determine arrow types to coerce data when creating batches
+        # Determine data types to coerce data when creating batches
         if isinstance(schema, StructType):
-            arrow_types = [to_arrow_type(f.dataType) for f in schema.fields]
+            data_types = [f.dataType for f in schema.fields]
         elif isinstance(schema, DataType):
             raise ValueError("Single data type %s is not supported with Arrow" % str(schema))
         else:
             # Any timestamps must be coerced to be compatible with Spark
-            arrow_types = [to_arrow_type(TimestampType())
-                           if is_datetime64_dtype(t) or is_datetime64tz_dtype(t) else None
-                           for t in pdf.dtypes]
+            data_types = [to_arrow_type(TimestampType())
+                          if is_datetime64_dtype(t) or is_datetime64tz_dtype(t) else None
+                          for t in pdf.dtypes]
 
         # Slice the DataFrame to be batched
         step = -(-len(pdf) // self.sparkContext.defaultParallelism)  # round int up
         pdf_slices = (pdf.iloc[start:start + step] for start in range(0, len(pdf), step))
 
         # Create list of Arrow (columns, type) for serializer dump_stream
-        arrow_data = [[(c, t) for (_, c), t in zip(pdf_slice.iteritems(), arrow_types)]
-                      for pdf_slice in pdf_slices]
+        # Type can be Spark SQL Data Type or Arrow Data Type
+        arrow_data_with_t = [
```

Comment:
this seems to be only a rename? should we leave them as they are to reduce the size of PR?

Reply:
Yes. It is renamed to indicate it is arrow data with datatype (Spark SQL DataType or Arrow DataType).

In `serializers.py`, `dt` is for Spark SQL DataType, `pdt` is for pyarrow DataType.

Reply:
Well, I should use `adt` or `padt` for PyArrow Data Type and `pdt` for Pandas DataType.